angular-workspace/projects/ni/nimble-angular/pipes/tests/duration.pipe.spec.ts

```diff
@@ -0,0 +1,46 @@
+import { DurationPipe } from '../duration.pipe';
+
+describe('DurationPipe', () => {
+    describe('basic formatting (English)', () => {
+        const pipe = new DurationPipe('en');
+
+        const testCases = [
+            {
```

Comment:
think we should have the ieee754 edge case tests -0, 0, NaN, Inf, -Inf, 

maybe also some boundary values max safe int, min safe int, max value, min value, epsilon

And the string behavior is new here, should probably exercise that more thoroughly. Angular seems pretty strict about not actual numbers being passed in. We should probably follow their behavior: https://github.com/angular/angular/blob/16.2.10/packages/common/src/pipes/number_pipe.ts#L293

Reply:
Added additional tests for numeric edge cases, and some additional string tests.

It looks like the Angular pipe [returns null](https://github.com/angular/angular/blob/16.2.10/packages/common/src/pipes/number_pipe.ts#L250) for `null` and `undefined`, but I think returning empty string as this pipe currently does makes more sense for our use cases (it's consistent with `DurationFormatter`, and the behavior of other pipes we have in SLE).  
Non-numeric string values will turn into empty string (`parseFloat` turns them into `NaN`, `DurationFormatter` makes `NaN` empty string). So since non-numeric strings are covered, and `transform` input is typed as `number|string|null|undefined`, I don't think there's any case we need to `throw` for.